_inc/client/state/connection/reducer.js

```diff
@@ -180,3 +180,57 @@ export function isFetchingUserData( state ) {
 export function isCurrentUserLinked( state ) {
 	return !! state.jetpack.connection.user.currentUser.isConnected;
 }
+
+/**
+ * Checks if the site is currently in development mode.
+ *
+ * @param  {Object}  state Global state tree
+ * @return {boolean} True if site is in dev mode. False otherwise.
+ */
+export function isDevMode( state ) {
+	return 'dev' === getSiteConnectionStatus( state );
+}
+
+/**
+ * Checks if the module requires connection.
+ *
+ * @param  {String}  slug Module slug.
+ * @return {boolean} True if module requires connection.
+ */
+export function requiresConnection( slug ) {
+	return [
+		// Modules
```

Comment:
this list isn't going to change anytime in the near future, and since it's just dev mode honestly I'm ok with a hardcoded version for this time around.